src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -188,10 +190,32 @@ private string GetRootPathWithGuid_NoLock()
             var fullName = GetFullReflectionName(containingOrThis);
 
             var compilation = await temporaryDocument.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-            // TODO: retrieve path to actual assembly instead of reference assembly
-            var reference = compilation.GetMetadataReference(symbol.ContainingAssembly);
+
+            string assemblyLocation = null;
+            var isReferenceAssembly = symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass.Name == nameof(ReferenceAssemblyAttribute));
+            if (isReferenceAssembly)
+            {
+                // Attempt to resolve the assembly via the current process' binding path
+                try
+                {
+                    var fullAssemblyName = symbol.ContainingAssembly.Identity.GetDisplayName();
+                    var loadedAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(x => x.FullName == fullAssemblyName);
+                    var assembly = loadedAssembly ?? Assembly.ReflectionOnlyLoad(fullAssemblyName);
+                    assemblyLocation = assembly.Location;
+                }
+                catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
```

Comment:
We shouldn't report NFW for load failures.

Reply:
💭 I was hoping it would provide insight into cases where Navigate to Decompiled Sources doesn't work properly, in order to prioritize future improvements. In light of this, can you confirm that this edge case should be removed from the telemetry consideration?